### PR TITLE
Restore unicase_support for phf_macros

### PR DESCRIPTION
**Pull request by gwenn**
See #182

### CODE REVIEW

File: phf_macros/Cargo.toml [hunk 1]
```diff
@@ -10,11 +10,15 @@ repository = "https://github.com/sfackler/rust-phf"
 [lib]
 proc-macro = true
 
+[features]
+unicase_support = ["unicase", "phf_shared/unicase"]
```
Comment (pickfire) [inline phf_macros/Cargo.toml:14]: Shouldn't we call it `unicase` for consistency?

Comment (gwenn) [reply phf_macros/Cargo.toml:14]: I am not sure a feature can have the same name as a dependency.

Comment (pickfire) [reply phf_macros/Cargo.toml:14]: Yes, it can. See the feature in `phf`.

Comment (gwenn) [reply phf_macros/Cargo.toml:14]: But there is no `unicase` dependency in `phf`.
And `unicase_support` is the original feature name: See https://github.com/sfackler/rust-phf/blob/master/phf_macros/tests/test.rs#L240 and https://github.com/sfackler/rust-phf/blob/master/phf_macros/tests/test.rs#L253

File: phf_macros/src/lib.rs [hunk 2]
```diff
@@ -116,6 +124,41 @@ impl ParsedKey {
...
             Expr::Group(group) => ParsedKey::from_expr(&group.expr),
+            #[cfg(feature = "unicase_support")]
+            Expr::Call(call) => match call.func.as_ref() {
+                Expr::Path(ep) => {
+                    if ep.path.leading_colon.is_none()
+                        && ep.path.segments.len() == 2
+                        && call.args.len() == 1
+                    {
+                        let first = ep.path.segments.first().unwrap();
+                        let last = ep.path.segments.last().unwrap();
```
Comment (JohnTitor) [inline phf_macros/src/lib.rs:135]:
```suggestion
                        let first = ep.path.segments.first()?;
                        let last = ep.path.segments.last()?;
```

File: phf_macros/src/lib.rs [hunk 3]
```diff
@@ -116,6 +124,41 @@ impl ParsedKey {
                 }
             }
             Expr::Group(group) => ParsedKey::from_expr(&group.expr),
+            #[cfg(feature = "unicase_support")]
+            Expr::Call(call) => match call.func.as_ref() {
```
Comment (JohnTitor) [inline phf_macros/src/lib.rs:128]: I'd prefer `if let` here.

File: phf_macros/Cargo.toml (hunk 1 above)
Comment (JohnTitor) [reply phf_macros/Cargo.toml:14]: I think it's okay to use the same name as previous 👍

**Comment by JohnTitor** [comment]
I opened #209 since this needs a rebase and some tweaks.

**Comment by gwenn** [comment]
> I opened #209 since this needs a rebase and some tweaks.

Thanks.

**Pull request closed without merging**